src/store/socket.js

```diff
@@ -18,19 +20,87 @@ export const runSocket = (token) => {
   socket = io(API.DOMAIN, {
     query: { token },
   });
-  socket.connect();
+
+  socket.on('connect', () => {
+
+  });
 
   socket.on('error', (e) => {
     if (e === 'Authentication error') {
       store.dispatch(refreshToken());
     }
   });
 
-  socket.on('new_message', ([message]) => {
-    store.dispatch(updateUserChat(message));
+  socket.on('DESTINATION_PUBLIC_KEY', ({ chatId, key }) => {
+    store.dispatch(setDestinationPublicKey(chatId, key));
+    getCurrentChat(chatId);
+  });
+
+  socket.on('SET_CURRENT_CHAT', (chat) => {
+    const { _id } = store.getState().user;
+    const { userChats } = store.getState().chats;
+    const privateKey = store.getState().messages.privateKeys[_id][chat._id];
+
+    const chatIdx = userChats
+      .findIndex(item => item._id === chat._id);
+    const updatedChats = [...userChats];
+    const decryptMessages = [...chat.messages];
+
+    if (decryptMessages.length > 0) {
```

Comment:
I think this `if` inner code you can move to separate function and the same thing for `else`
```
const decryptMessagesPromises = decryptMessages
        .map(message => Promise.resolve(decrypt(message.text, privateKey))
          .then(r => ({  ...message,
            text: r,
          })));

      Promise.all(decryptMessagesPromises).then((dec) => {
        if (chatIdx > -1) {
          updatedChats[chatIdx] = { ...chat, messages: dec.reverse() };
        } else {
          updatedChats.push(chat);
        }

        store.dispatch(setChat({ ...chat, messages: dec }));
        store.dispatch(setChats(updatedChats));
        store.dispatch(setPending(false));
        store.dispatch(navigate(ROUTES.CHAT));
      });
```